Cannot build the vue3-sample-app

### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.26.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

2.18.0

### Public or Confidential Client?

Public

### Description

reg, samples/msal-browser-samples/vue3-sample-app

i cannot build the vue3-sample-app with following error.

can anyone help me for this issue?

`
$ npm run build

> msal-vue-sample@0.0.0 build /Users/takuya/microsoft-authentication-library-for-js/samples/msal-browser-samples/vue3-sample-app
> vue-tsc --noEmit && vite build

src/views/Profile.vue:2:14 - error TS2304: Cannot find name 'state'.

2   <div v-if="state.resolved">
               ~~~~~

src/views/Home.vue:3:18 - error TS2304: Cannot find name 'isAuthenticated'.

3     <span v-if="!isAuthenticated">Please sign-in to see your profile information.</span>
                   ~~~~~~~~~~~~~~~

src/views/Profile.vue:3:14 - error TS2304: Cannot find name 'state'.

3  <p>Name: {{ state.data.displayName }}</p>

:

node_modules/element-plus/es/utils/objects.d.ts:1:30 - error TS2307: Cannot find module 'type-fest' or its corresponding type declarations.

1 import type { Entries } from 'type-fest';
                               ~~~~~~~~~~~


Found 17 errors in 9 files.

Errors  Files
     6  src/views/Profile.vue:2
     2  src/views/Home.vue:3
     1  src/components/WelcomeName.vue:3
     2  src/components/SignOutButton.vue:6
     2  src/components/SignInButton.vue:6
     1  src/components/NavBar.vue:13
     1  node_modules/element-plus/es/components/table-v2/src/components/sort-icon.d.ts:5
     1  node_modules/element-plus/es/components/table-v2/src/types.d.ts:92
     1  node_modules/element-plus/es/utils/objects.d.ts:1
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! msal-vue-sample@0.0.0 build: `vue-tsc --noEmit && vite build`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the msal-vue-sample@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/takuya/.npm/_logs/2022-10-20T10_45_54_786Z-debug.log
`


### MSAL Configuration

_No response_

### Relevant Code Snippets

_No response_

### Identity Provider

_No response_

### Source

External (Customer)

--- a/samples/msal-browser-samples/vue3-sample-app/README.md
+++ b/samples/msal-browser-samples/vue3-sample-app/README.md
@@ -2,7 +2,7 @@
 
 ## About this sample
 
-This sample demonstrates one way you can integrate the `@azure/msal-browser` package into your Vue 3 application using the [composition API](https://v3.vuejs.org/api/composition-api.html). It is not exhaustive and there may simpler or more complex solutions depending on your specific use case.
+This sample demonstrates one way you can integrate the `@azure/msal-browser` package into your Vue 3 application using the [composition API](https://v3.vuejs.org/api/composition-api.html). It is not exhaustive and there may be simpler or more complex solutions depending on your specific use case.
 
 ⚠️ This sample is currently for demonstration purposes only. Support will be limited.
 
@@ -47,9 +47,10 @@ This sample demonstrates how you can integrate `@azure/msal-browser` into your V
 
 You'll find the MSAL configuration and `PublicClientApplication` instantiation in `authConfig.ts`. It's very important that `PublicClientApplication` is initialized only once per pageload and as such should not be initialized inside any Vue components, but rather outside the context of Vue and passed in.
 
-The `clientId` is the most important, and only required parameter, as it maps to your app registration in the Azure Portal.
-The `authority` represents the Azure AD instance and tenant that MSAL.js will use the sign users in. This parameter controls the audience of your app.
-The `redirectUri` and `postLogoutRedirectUri` represent where AAD will redirect you back to after logging in and must be registered on your app registration as type "SPA". If you do not provide these, MSAL.js will use the current page by default.
+- The `clientId` is the most important, and only required parameter, as it maps to your app registration in the Azure Portal.
+- The `authority` represents the Azure AD instance and tenant that MSAL.js will use the sign users in. This parameter controls the audience of your app.
+    > :information_source: If you need to sign users in with Azure AD B2C, this parameter should be set to the [B2C tenanted authority](../../../lib/msal-common/docs/authority.md#azure-ad-b2c) with the default user-flow/custom policy that will be used for sign-ins and token acquisitions. To learn more about how to handle B2C user-flows and/or custom policies with MSAL.js, please refer to [react-b2c-sample](../../msal-react-samples/b2c-sample/) and/or [angular-b2c-sample](../../msal-angular-v2-samples/angular-b2c-sample-app/).
+- The `redirectUri` and `postLogoutRedirectUri` represent where AAD will redirect you back to after logging in and must be registered on your app registration as type "SPA". If you do not provide these, MSAL.js will use the current page by default.
 
 The `cacheLocation` configures where you want your tokens to be stored. SessionStorage is the default, if this option is not provided.
 

--- a/samples/msal-browser-samples/vue3-sample-app/README.md
+++ b/samples/msal-browser-samples/vue3-sample-app/README.md
@@ -2,7 +2,7 @@
 
 ## About this sample
 
-This sample demonstrates one way you can integrate the `@azure/msal-browser` package into your Vue 3 application using the [composition API](https://v3.vuejs.org/api/composition-api.html). It is not exhaustive and there may simpler or more complex solutions depending on your specific use case. 
+This sample demonstrates one way you can integrate the `@azure/msal-browser` package into your Vue 3 application using the [composition API](https://v3.vuejs.org/api/composition-api.html). It is not exhaustive and there may simpler or more complex solutions depending on your specific use case.
 
 ⚠️ This sample is currently for demonstration purposes only. Support will be limited.
 
@@ -48,6 +48,7 @@ This sample demonstrates how you can integrate `@azure/msal-browser` into your V
 You'll find the MSAL configuration and `PublicClientApplication` instantiation in `authConfig.ts`. It's very important that `PublicClientApplication` is initialized only once per pageload and as such should not be initialized inside any Vue components, but rather outside the context of Vue and passed in.
 
 The `clientId` is the most important, and only required parameter, as it maps to your app registration in the Azure Portal.
+The `authority` represents the Azure AD instance and tenant that MSAL.js will use the sign users in. This parameter controls the audience of your app.
 The `redirectUri` and `postLogoutRedirectUri` represent where AAD will redirect you back to after logging in and must be registered on your app registration as type "SPA". If you do not provide these, MSAL.js will use the current page by default.
 
 The `cacheLocation` configures where you want your tokens to be stored. SessionStorage is the default, if this option is not provided.
@@ -56,8 +57,9 @@ The `cacheLocation` configures where you want your tokens to be stored. SessionS
 const msalConfig = {
   auth: {
     clientId: 'ENTER_YOUR_CLIENT_ID_HERE',
-    redirectUri: 'http://localhost:3000', // Must be registered as a SPA redirectURI on your app registration
-    postLogoutRedirectUri: 'http://localhost:3000', // Must be registered as a SPA redirectURI on your app registration
+    authority: 'https://login.microsoftonline.com/ENTER_YOUR_TENANT_ID_HERE',
+    redirectUri: '/', // Must be registered as a SPA redirectURI on your app registration
+    postLogoutRedirectUri: '/' // Must be registered as a SPA redirectURI on your app registrationregistration
   },
   cache: {
     cacheLocation: 'localStorage' // Options are localStorage, sessionStorage, memoryStorage

--- a/samples/msal-browser-samples/vue3-sample-app/package.json
+++ b/samples/msal-browser-samples/vue3-sample-app/package.json
@@ -4,10 +4,13 @@
   "scripts": {
     "start": "vite",
     "build": "vue-tsc --noEmit && vite build",
-    "serve": "vite preview"
+    "serve": "vite preview",
+    "build:package": "cd ../../../lib/msal-browser && npm run build:all",
+    "install:local": "npm install ../../../lib/msal-browser",
+    "install:published": "npm install @azure/msal-browser@latest"
   },
   "dependencies": {
-    "@azure/msal-browser": "^2.26.0",
+    "@azure/msal-browser": "file:../../../lib/msal-browser",
     "element-plus": "^2.2.19",
     "vue": "^3.2.41",
     "vue-router": "^4.1.5"

--- a/samples/msal-browser-samples/vue3-sample-app/package.json
+++ b/samples/msal-browser-samples/vue3-sample-app/package.json
@@ -10,7 +10,7 @@
     "install:published": "npm install @azure/msal-browser@latest"
   },
   "dependencies": {
-    "@azure/msal-browser": "file:../../../lib/msal-browser",
+    "@azure/msal-browser": "^2.30.0",
     "element-plus": "^2.2.19",
     "vue": "^3.2.41",
     "vue-router": "^4.1.5"

--- a/samples/msal-browser-samples/vue3-sample-app/package.json
+++ b/samples/msal-browser-samples/vue3-sample-app/package.json
@@ -8,14 +8,14 @@
   },
   "dependencies": {
     "@azure/msal-browser": "^2.26.0",
-    "element-plus": "^2.2.6",
-    "vue": "^3.2.37",
-    "vue-router": "^4.0.16"
+    "element-plus": "^2.2.19",
+    "vue": "^3.2.41",
+    "vue-router": "^4.1.5"
   },
   "devDependencies": {
-    "@vitejs/plugin-vue": "^1.9.3",
-    "typescript": "^4.4.3",
-    "vite": "^2.6.4",
-    "vue-tsc": "^0.3.0"
+    "@vitejs/plugin-vue": "^3.1.2",
+    "typescript": "^4.8.4",
+    "vite": "^3.1.8",
+    "vue-tsc": "^1.0.8"
   }
 }

--- a/samples/msal-browser-samples/vue3-sample-app/src/authConfig.ts
+++ b/samples/msal-browser-samples/vue3-sample-app/src/authConfig.ts
@@ -4,8 +4,9 @@ import { LogLevel, PublicClientApplication } from '@azure/msal-browser';
 export const msalConfig = {
   auth: {
     clientId: 'ENTER_YOUR_CLIENT_ID_HERE',
-    redirectUri: 'http://localhost:3000', // Must be registered as a SPA redirectURI on your app registration
-    postLogoutRedirectUri: 'http://localhost:3000' // Must be registered as a SPA redirectURI on your app registration
+    authority: 'https://login.microsoftonline.com/ENTER_YOUR_TENANT_ID_HERE',
+    redirectUri: '/', // Must be registered as a SPA redirectURI on your app registration
+    postLogoutRedirectUri: '/' // Must be registered as a SPA redirectURI on your app registration
   },
   cache: {
     cacheLocation: 'localStorage'
@@ -13,22 +14,22 @@ export const msalConfig = {
   system: {
       loggerOptions: {
           loggerCallback: (level: LogLevel, message: string, containsPii: boolean) => {
-              if (containsPii) {	
-                  return;	
+              if (containsPii) {
+                  return;
               }
-              switch (level) {	
-                  case LogLevel.Error:	
-                      console.error(message);	
-                      return;	
-                  case LogLevel.Info:	
-                      console.info(message);	
-                      return;	
-                  case LogLevel.Verbose:	
-                      console.debug(message);	
-                      return;	
-                  case LogLevel.Warning:	
-                      console.warn(message);	
-                      return;	
+              switch (level) {
+                  case LogLevel.Error:
+                      console.error(message);
+                      return;
+                  case LogLevel.Info:
+                      console.info(message);
+                      return;
+                  case LogLevel.Verbose:
+                      console.debug(message);
+                      return;
+                  case LogLevel.Warning:
+                      console.warn(message);
+                      return;
                   default:
                       return;
               }

--- a/samples/msal-browser-samples/vue3-sample-app/src/utils/UserInfo.ts
+++ b/samples/msal-browser-samples/vue3-sample-app/src/utils/UserInfo.ts
@@ -1,0 +1,18 @@
+/**
+ * Graph data about the user.
+ */
+type UserInfo = {
+    businessPhones?: Array<string>,
+    displayName?: string,
+    givenName?: string,
+    id?: string,
+    jobTitle?: string,
+    mail?: string,
+    mobilePhone?: string,
+    officeLocation?: string,
+    preferredLanguage?: string,
+    surname?: string,
+    userPrincipalName?: string
+};
+
+export default UserInfo;

--- a/samples/msal-browser-samples/vue3-sample-app/src/views/Profile.vue
+++ b/samples/msal-browser-samples/vue3-sample-app/src/views/Profile.vue
@@ -3,7 +3,7 @@
 	<p>Name: {{ state.data.displayName }}</p>
 	<p>Title: {{ state.data.jobTitle }}</p>
 	<p>Mail: {{ state.data.mail }}</p>
-	<p>Phone: {{ state.data.businessPhones[0] }}</p>
+	<p>Phone: {{ state.data.businessPhones ? state.data.businessPhones[0] : "" }}</p>
 	<p>Location: {{ state.data.officeLocation }}</p>
   </div>
 </template>
@@ -14,12 +14,13 @@ import { InteractionRequiredAuthError, InteractionStatus } from "@azure/msal-bro
 import { reactive, onMounted, watch } from 'vue'
 import { loginRequest } from "../authConfig";
 import { callMsGraph } from "../utils/MsGraphApiCall";
+import UserInfo from "../utils/UserInfo";
 
 const { instance, inProgress } = useMsal();
 
 const state = reactive({
 	resolved: false,
-	data: {}
+	data: {} as UserInfo
 });
 
 async function getGraphData() {

--- a/samples/msal-browser-samples/vue3-sample-app/src/views/ProfileNoGuard.vue
+++ b/samples/msal-browser-samples/vue3-sample-app/src/views/ProfileNoGuard.vue
@@ -3,7 +3,7 @@
 	<p>Name: {{ state.data.displayName }}</p>
 	<p>Title: {{ state.data.jobTitle }}</p>
 	<p>Mail: {{ state.data.mail }}</p>
-	<p>Phone: {{ state.data.businessPhones[0] }}</p>
+	<p>Phone: {{ state.data.businessPhones ? state.data.businessPhones[0] : "" }}</p>
 	<p>Location: {{ state.data.officeLocation }}</p>
   </div>
 </template>
@@ -14,12 +14,13 @@ import { InteractionType } from "@azure/msal-browser";
 import { reactive, watch } from 'vue'
 import { loginRequest } from "../authConfig";
 import { callMsGraph } from "../utils/MsGraphApiCall";
+import UserInfo from "../utils/UserInfo";
 
 const { result, acquireToken } = useMsalAuthentication(InteractionType.Redirect, loginRequest);
 
 const state = reactive({
 	resolved: false,
-	data: {}
+	data: {} as UserInfo
 });
 
 async function getGraphData() {

--- a/samples/msal-browser-samples/vue3-sample-app/tsconfig.json
+++ b/samples/msal-browser-samples/vue3-sample-app/tsconfig.json
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "skipLibCheck": true,
     "target": "esnext",
     "useDefineForClassFields": true,
     "module": "esnext",

--- a/samples/msal-browser-samples/vue3-sample-app/vite.config.ts
+++ b/samples/msal-browser-samples/vue3-sample-app/vite.config.ts
@@ -3,5 +3,9 @@ import vue from '@vitejs/plugin-vue'
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [vue()]
+  plugins: [vue()],
+    server: {
+        host: 'localhost',
+        port: 3000,
+    },
 })

--- a/samples/msal-browser-samples/vue3-sample-app/vite.config.ts
+++ b/samples/msal-browser-samples/vue3-sample-app/vite.config.ts
@@ -1,5 +1,8 @@
-import { defineConfig } from 'vite'
-import vue from '@vitejs/plugin-vue'
+import { defineConfig } from 'vite';
+import vue from '@vitejs/plugin-vue';
+import dns from 'dns';
+
+dns.setDefaultResultOrder('verbatim');
 
 // https://vitejs.dev/config/
 export default defineConfig({

